rust/template/src/api/c_api.rs

```diff
@@ -35,45 +35,88 @@ use std::os::unix::io::{FromRawFd, IntoRawFd, RawFd};
 use std::os::windows::io::{FromRawHandle, IntoRawHandle, RawHandle};
 
 #[no_mangle]
-pub unsafe extern "C" fn ddlog_get_table_id(tname: *const raw::c_char) -> libc::size_t {
-    if tname.is_null() {
+pub unsafe extern "C" fn ddlog_get_table_id(
+    prog: *const HDDlog,
```

Comment:
To be truly FFI safe while still allowing consumers to take advantage of ddlog we probably want the type used by consumers to be an `Arc<Box<dyn DDlog>>` or similar so that any ddlog runtime can be used while keeping it thread safe since dynamically dispatched objects use fat pointers which aren't ffi safe